crypto/storage/storage.go

```diff
@@ -20,16 +20,65 @@ package storage
 
 import (
 	"crypto"
+	"encoding/json"
 	"errors"
+	"fmt"
+
+	"github.com/lestrrat-go/jwx/jwk"
+	"github.com/nuts-foundation/nuts-node/core"
+	"github.com/nuts-foundation/nuts-node/crypto/util"
 )
 
 // ErrNotFound indicates that the specified crypto storage entry couldn't be found.
 var ErrNotFound = errors.New("entry not found")
 
-// Storage interface containing functions for storing and retrieving keys
+// Storage interface containing functions for storing and retrieving keys.
 type Storage interface {
 	GetPrivateKey(kid string) (crypto.Signer, error)
-	GetPublicKey(kid string) (crypto.PublicKey, error)
 	PrivateKeyExists(kid string) bool
 	SavePrivateKey(kid string, key crypto.PrivateKey) error
+
+	// GetPublicKey returns the public key and the period it is valid in.
+	GetPublicKey(kid string) (PublicKeyEntry, error)
+	// SavePublicKey stores a public key entry under the given kid
+	SavePublicKey(kid string, entry PublicKeyEntry) error
+}
+
+// PublicKeyEntry is a public key entry also containing the period it's valid for.
+type PublicKeyEntry struct {
+	Period       core.Period `json:"period"`
+	parsedJWK    jwk.Key
+	PublicKeyJwk map[string]interface{} `json:"publicKeyJwk,omitempty"`
```

Comment:
does it need to be named 'PublicKeyJwk', since we're already in `PublicKeyEntry`? It can just be named `JWK` or `Key`?

Reply:
used JWK, but left it as publicKeyJwk in json